src/renderer.js

```diff
@@ -110,7 +114,22 @@ Renderer.prototype.resize = function (resolution) {
   return false;
 };
 
-function updateUniforms(gl, camera, uniforms) {
+Renderer.prototype.texturize = function (image) {
+  if (!image) {
+    this.texture = null;
+    this.glTexture = createTexture(this.gl, block);
+    return Promise.resolve();
+  }
+
+  return getImage(image)
+    .then(function (pixels) {
+      this.texture = image;
+      this.glTexture = createTexture(this.gl, pixels);
+    }.bind(this))
+    .catch(function (err) { console.log(err); });
+};
```

Comment:
I'm not sure if WebGL will automatically free up the old textures. If not, switching back and forth between textures will cause a memory leak. You may want to consider using deleteTexture in a future revision.